### PR TITLE
Support harvester cloud provider

### PR DESCRIPTION
**Pull request by yaocw2020**
Related issue: 
https://github.com/rancher/rancher/issues/35000
https://github.com/harvester/harvester/issues/1396

### CODE REVIEW

File: types/rke_types.go
Comment (gitlawr) [inline types/rke_types.go]: I think there should be related generated codes for this change?

Comment (yaocw2020) [reply types/rke_types.go]: Yes, we should bump the rke package in Rancher.

Comment (guangbochen) [reply types/rke_types.go]: I think lawr is referring to the generated files in this repo, e.g https://github.com/rancher/rke/blob/master/types/zz_generated_deepcopy.go

Comment (yaocw2020) [reply types/rke_types.go]: Rancher refers rke/types here.  https://github.com/rancher/rancher/blob/ea7823e463847ac1faf3dfdd62dbab7eec48bde1/pkg/apis/management.cattle.io/v3/cluster_types.go#L109

Comment (yaocw2020) [reply types/rke_types.go]: There should be the related generated codes, but I can't generate code by the command `go generate`. I am still figuring out the reason.

Comment (yaocw2020) [reply types/rke_types.go]: `HarvesterCloudProvider` is a string so it does not need a deepcopy method. @gitlawr @guangbochen

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by nickgerace — APPROVED** [review]

File: types/rke_types.go
Comment (StrongMonkey) [reply types/rke_types.go]: Can this be a go struct so that it match the format with other cloud providers? Any reason to use string here? Also you get a benefit of adding more fields in harvester cloud provider later without having to tweak the api.

**Review by StrongMonkey — CHANGES_REQUESTED** [review]
Can this be a go struct so that it match the format with other cloud providers? Any reason to use string here? Also you get a benefit of adding more fields in harvester cloud provider later without having to tweak the api.

**Comment by yaocw2020** [comment]
> Can this be a go struct so that it match the format with other cloud providers? Any reason to use string here? Also you get a benefit of adding more fields in harvester cloud provider later without having to tweak the api.

The cloud config for the Harvester cloud provider is actually a kubeconfig generated by calling the Harvester api.

**Comment by StrongMonkey** [comment]
But can you actually make it a go struct similar to other cloud providers? Just for consistency and extensibility.

**Comment by guangbochen** [comment]
@yaocw2020 run the `go generate` should be required with the current struct type change.

**Comment by yaocw2020** [comment]
`go generate` will be functional if the project codes are under the go src path.

**Review by StrongMonkey — APPROVED** [review]

**Review by nickgerace — APPROVED** [review]
@StrongMonkey I think we are good to merge, right?

**Comment by guangbochen** [comment]
Hey folks, wondering if we should also submit the go mod bump on the `rancher/rancher` after PR merge? thanks in advance.

**Comment by nickgerace** [comment]
@guangbochen yes. I think opening a PR now there could be helpful (`release/v2.6`)

**Comment by StrongMonkey** [comment]
cc @guangbochen assigning @kinarashah as she is the committer and can help coordinate the merge.

**Review by kinarashah — APPROVED** [review]
@guangbochen Looks good, do you need RKE CLI RC for this? Lemme know and I can tag one. Otherwise, can just pull in the commit in Rancher go.mod.

**Comment by guangbochen** [comment]
CLI is not necessary for now, just bump the go.mod should be good enough, thanks.

**Pull request merged**
